core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -994,6 +1006,139 @@ public void date_format() {
     assertEquals(missingValue(), eval(dsl.date_format(DSL.literal(""), missingRef)));
   }
 
+  private FunctionExpression maketime(Expression hour, Expression minute, Expression second) {
+    var repo = new ExpressionConfig().functionRepository();
+    var func = repo.resolve(new FunctionSignature(new FunctionName("maketime"),
+        List.of(DOUBLE, DOUBLE, DOUBLE)));
+    return (FunctionExpression)func.apply(List.of(hour, minute, second));
+  }
+
+  private LocalTime maketime(Double hour, Double minute, Double second) {
+    return maketime(DSL.literal(hour), DSL.literal(minute), DSL.literal(second))
+        .valueOf(null).timeValue();
+  }
+
+  @Test
+  public void maketime() {
+    var r = new Random();
+    assertEquals(maketime(20., 30., 40.), LocalTime.of(20, 30, 40));
+
+    for (var ignored : IntStream.range(0, 25).toArray()) {
```

Comment:
Why is this for loop necessary? What does it test that isn't covered by the test on line 1024 above?

As an aside, tests must be reproducible so if a test uses RNG, the user must be able to seed it and any test failures must include the seed used. Otherwise, if it does fail, there is no way to investigate the failure.

Reply:
Instead of testing RNG numbers, test boundary cases.  For example: 0, and 23:59:59.99 would be decent test cases for time. 
Feb 29 on leap years, and older years would be good boundary cases for dates. 
Lots of timezones (and over the the dateline) are also excellent test cases when timezones are included.

Reply:
I dump values used on case of failure, but if it not enough I can hardcode values gotten from RNG.